cmd/authctl/user/disable.go

```diff
@@ -0,0 +1,34 @@
+package user
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/ubuntu/authd/internal/consts"
+	"github.com/ubuntu/authd/internal/proto/authd"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// DisableCmd is a command to disable a user.
+var DisableCmd = &cobra.Command{
+	Use:   "disable",
+	Short: "Disable a user managed by authd",
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		fmt.Printf("Disabling user %s\n", args[0])
+		conn, err := grpc.NewClient("unix://"+consts.DefaultSocketPath, grpc.WithTransportCredentials(insecure.NewCredentials()))
+		if err != nil {
+			return err
+		}
+
```

Comment:
Also check for the HealthCheck to be around? See pam adapter model but it can be simpler here.

Reply:
Ah, do not make the request when the socket is there but not ready (so in a socket-activated scenario) or in tests where the daemon may take some time to show up.

Reply:
Mmh, I don't really understand how calling the `Health.Check` RPC method instead of the `NSS.DisableUser` RPC method would help us when the daemon is not ready. Both have a timeout and if the daemon is ready before the timeout then they succeed, otherwise they fail.

My understanding of the gRPC health checks is that they are useful for two things: monitoring the status of gRPC services and choosing a healthy service from a list of services.

Reply:
Well it allows to wait properly for the connection to be up, without having to call a method that may return a false-positive error (because things are still loading).

However, in the past I had handled this for all so here the only thing to call is basically `grpcutils.WaitForConnection(context.TODO(), conn, 1*time.Second)` and it will do everything, and if we fail (mostly on tests where we may be slower) we've a clear reason.

Reply:
No, it's not different but unlikely the others it's a safe no-op that we can use in order to ensure that a specific service is active. In our case, all the services are handled together so it's not needed to handle this per-service, but having it was important in order to be able to check and wait for the daemon availability without any side-effect.

And it was introduced to handle properly cases in which authd is not yet ready or goes down (see #694).

So in this case, instead of trying to call `DisableUser` until we're available, it's just better to instead wait for authd availability (really, it's an one-liner change) and then go for it, and this will pay back in integration tests where a request may happen just after authd is started but the service may not be yet ready.

Reply:
Differently to other calls, it's a no-op, and it allows to repeatedly check for the demon for a specified time, before than failing.

So it gives some time to the daemon to  start the socket and setup all the services, then it fails if it can't get there. While a normal call would just fail